From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sr-0w <Sr-0w@users.noreply.github.com>
Date: Sat, 8 Aug 2026 00:45:00 +0200
Subject: [PATCH] arm64: dts: qcom: hotdog: add the camera PMIC digital rails

PM8009 is the dedicated camera PMIC and is already described, but only three
of its LDOs are, and none of them are the ones the image sensors use.

The stock device tree drives the sensors' digital supply from ldo1, ldo3 and
ldo4, at 1.104 V for the slot on CSIPHY1 and 1.056 V for the other three.
Describe them so a sensor node has something to reference.

Signed-off-by: Sr-0w <Sr-0w@users.noreply.github.com>
---
 arch/arm64/boot/dts/qcom/sm8150-oneplus-common.dtsi | 22 ++++++++++
 1 file changed, 22 insertions(+)

--- a/arch/arm64/boot/dts/qcom/sm8150-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8150-oneplus-common.dtsi
@@ -394,15 +394,36 @@
 		vdd-s1-supply = <&vph_pwr>;
 		vdd-s2-supply = <&vreg_bob>;
 
+		vdd-l1-supply = <&vreg_s8c_1p3>;
 		vdd-l2-supply = <&vreg_s8c_1p3>;
+		vdd-l3-supply = <&vreg_s8c_1p3>;
+		vdd-l4-supply = <&vreg_s8c_1p3>;
 		vdd-l5-l6-supply = <&vreg_bob>;
 
+		vreg_l1f_1p1: ldo1 {
+			regulator-min-microvolt = <1104000>;
+			regulator-max-microvolt = <1104000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
 		vreg_l2f_1p2: ldo2 {
 			regulator-min-microvolt = <1200000>;
 			regulator-max-microvolt = <1200000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l3f_1p05: ldo3 {
+			regulator-min-microvolt = <1056000>;
+			regulator-max-microvolt = <1056000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
+		vreg_l4f_1p05: ldo4 {
+			regulator-min-microvolt = <1056000>;
+			regulator-max-microvolt = <1056000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
 		vreg_l5f_2p85: ldo5 {
 			regulator-min-microvolt = <2800000>;
 			regulator-max-microvolt = <2800000>;
--
2.50.1
